backend/persistent/QueryScheduler.js

```diff
@@ -381,11 +387,17 @@ class QueryScheduler {
                 */
                 throw new Error(`MetadataError: ${e.message}`);
             });
-        }).then((res) => {
+        }).then(() => {
             Logger.log(`Request to Plotly for creating a grid took ${process.hrtime(startTime)[0]} seconds`, 2);
-            return res.json().then(() => {
-                Logger.log(`Grid ${fid} has been updated.`, 2);
-            });
+            Logger.log(`Grid ${fid} has been updated.`, 2);
+
+            startTime = process.hrtime();
+
+            // fetch updated grid for returning
+            return getGrid(fid, requestor);
+        }).then((res) => {
+            Logger.log(`Request to Plotly for fetching updated grid took ${process.hrtime(startTime)[0]} seconds`, 2);
+            return res.json();
```

Comment:
Test status before converting to JSON, otherwise the error will be reported as a failure to parse into JSON.